### PR TITLE
(Do not look at this)(Seriously) Flathub publishing updates

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
### Description

We need to pass the build-log-url now, and validate our output with the flatpak-builder-linter tool. Let's do that.

### Motivation and Context

Flathub is changing, and so should we.

### How Has This Been Tested?

It hasn't.

### Types of changes

 - Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: .github/workflows/build-project.yaml [hunk 1]
```diff
@@ -286,8 +286,13 @@ jobs:
 
           echo "cacheKey=${cache_key}" >> $GITHUB_OUTPUT
 
+      - name: Validate Flatpak manifest
+        run: |
+          : Validate Flatpak manifest
+          flatpak run --command=flatpak-builder-lint org.flatpak.Builder manifest ${{ github.workspace }}/build-aux/com.obsproject.Studio.json
```
Comment (tytan652) [inline .github/workflows/build-project.yaml:292]: I don't recommend doing this, the JSON schema provided by the linter is not up to date against the one in the flatpak/flatpak-builder repo.

Comment (GeorgesStavracas) [reply .github/workflows/build-project.yaml:292]: We should just follow the most recent one

Comment (tytan652) [reply .github/workflows/build-project.yaml:292]: You also need to install the Builder Flatpak, like I did in my WIP commit https://github.com/tytan652/obs-studio/commit/007bf8512d35820089f0ae583f9694eea2da804a

**Comment by tytan652** [comment]
Superseded by #9814

**Pull request closed without merging**